File: js/import-content-items/UpsertLanguageVariant.js
// DocSection: importing_upsert_variant
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
// Using ES6 syntax
import { ManagementClient } from '@kentico/kontent-management';

const client = new ManagementClient({
  projectId: '<YOUR_PROJECT_ID>',
  apiKey: '<YOUR_MANAGEMENT_API_KEY>'
});

const response = await client.upsertLanguageVariant()
  .byItemExternalId('ext-cafe-brno')
  .byLanguageCodename('en-US')
  .withData((builder) => [
    builder.textElement({
      element: {
        codename: 'street'
      },
      value: 'Nove Sady 25'
    }),
    builder.textElement({
      element: {
        codename: 'city'
      },
      value: 'Brno'
    }),
    builder.textElement({
      element: {
        codename: 'country'
      },
      value: 'Czech Republic'
    }),
    builder.textElement({
      element: {
        codename: 'state'
      },
      value: 'Jihomoravsky kraj'
    }),
    builder.textElement({
      element: {
        codename: 'zip_code'
      },
      value: '60200'
    }),
    builder.textElement({
      element: {
        codename: 'phone'
      },
      value: '+420555555555'
    }),
    builder.textElement({
      element: {
        codename: 'email'
      },
      value: 'brnocafe@kontent.ai'
    })
  ])
  .toPromise();
// EndDocSection

File: net/import-content-items/UpsertLanguageVariant.cs
// DocSection: importing_upsert_variant
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kentico.Kontent.Management;

var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "<YOUR_API_KEY>",
    ProjectId = "<YOUR_PROJECT_ID>"
});

var response = await client.UpsertLanguageVariantAsync(identifier, new LanguageVariantUpsertModel
{
    Elements = ElementBuilder.GetElementsAsDynamic(new BaseElement[]
    {
        new TextElement
        {
            Element = Reference.ByExternalId("street"),
            Value = "Nove Sady 25",
        },
        new TextElement
        {
            Element = Reference.ByExternalId("city"),
            Value = "Brno",
        },
        new TextElement
        {
            Element = Reference.ByExternalId("country"),
            Value = "Czech republic",
        },
        new TextElement
        {
            Element = Reference.ByExternalId("state"),
            Value = "Jihomoravsky kraj",
        },
        new TextElement
        {
            Element = Reference.ByExternalId("zip_code"),
            Value = "60200",
        },
        new TextElement
        {
            Element = Reference.ByExternalId("phone"),
            Value = "+420 555 555 555",
        },
        new TextElement
        {
            Element = Reference.ByExternalId("email"),
            Value = "brnocafe@kontent.ai",
        },
    })
});
// EndDocSection